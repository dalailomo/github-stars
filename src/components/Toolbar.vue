<template>
  <div>
    <v-navigation-drawer
      floating
      v-model="drawer"
      app
    >
      <StarredList />
    </v-navigation-drawer>
    <v-toolbar fixed app>
      <v-toolbar-side-icon :disabled="items.length < 1" @click.stop="drawer = !drawer"></v-toolbar-side-icon>
      <v-toolbar-title><UserSearch /></v-toolbar-title>
    </v-toolbar>
  </div>
</template>

<script>
import { mapState } from 'vuex'
import StarredList from '@/components/Toolbar/StarredList'
import UserSearch from '@/components/Toolbar/UserSearchForm'

export default {
  components: {
    StarredList,
    UserSearch,
  },

  computed: {
    ...mapState({
      items: state => state.starredList.items,
    }),
  },

  data: () => ({
    drawer: false,
  }),

  props: {
    source: {
      required: false,
      type: String,
      default: '',
    },
  },
}
</script>
