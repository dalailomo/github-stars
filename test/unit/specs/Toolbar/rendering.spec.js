import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Toolbar from '@/components/Toolbar'
import { getWrapper } from './../../helpers'
import { emptyState, stateWithItems } from './mocks'

Vue.use(Vuex)
Vue.use(Vuetify)

describe('Toolbar rendering', () => {
  it('Renders toolbar with disabled drawer when there are no items', () => {
    const wrapper = getWrapper(Toolbar, {}, emptyState, {}, {})

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Renders toolbar with enabled drawer when there are items', () => {
    const wrapper = getWrapper(Toolbar, {}, stateWithItems, {}, {})

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
