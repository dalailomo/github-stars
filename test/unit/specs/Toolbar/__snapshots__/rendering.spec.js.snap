// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar rendering Renders toolbar with disabled drawer when there are no items 1`] = `
<div>
  <aside class="navigation-drawer navigation-drawer--close navigation-drawer--fixed navigation-drawer--floating" style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); width: 300px;">
    <!---->
    <div class="navigation-drawer__border"></div>
  </aside>
  <nav class="toolbar toolbar--fixed" style="margin-top: 0px; padding-right: 0px; padding-left: 0px;">
    <div class="toolbar__content" style="height: 56px;">
      <button disabled="disabled" type="button" class="toolbar__side-icon btn btn--disabled btn--icon">
        <div class="btn__content">
          <i aria-hidden="true" class="icon material-icons">menu</i>
        </div>
      </button>
      <div class="toolbar__title">
        <!---->
      </div>
    </div>
  </nav>
</div>
`;

exports[`Toolbar rendering Renders toolbar with enabled drawer when there are items 1`] = `
<div>
  <aside class="navigation-drawer navigation-drawer--close navigation-drawer--fixed navigation-drawer--floating" style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); width: 300px;">
    <!---->
    <div class="navigation-drawer__border"></div>
  </aside>
  <nav class="toolbar toolbar--fixed" style="margin-top: 0px; padding-right: 0px; padding-left: 0px;">
    <div class="toolbar__content" style="height: 56px;">
      <button type="button" class="toolbar__side-icon btn btn--icon">
        <div class="btn__content">
          <i aria-hidden="true" class="icon material-icons">menu</i>
        </div>
      </button>
      <div class="toolbar__title">
        <!---->
      </div>
    </div>
  </nav>
</div>
`;
